controllers/components/dashboard/dashboard.go

```diff
@@ -87,22 +85,22 @@ func (s *componentHandler) UpdateDSCStatus(dsc *dscv1.DataScienceCluster, obj cl
 		dsc.Status.InstalledComponents[LegacyComponentNameUpstream] = true
 		dsc.Status.Components.Dashboard.DashboardCommonStatus = c.Status.DashboardCommonStatus.DeepCopy()
 
-		if rc := meta.FindStatusCondition(c.Status.Conditions, status.ConditionTypeReady); rc != nil {
-			nc.Status = corev1.ConditionStatus(rc.Status)
+		if rc := conditions.FindStatusCondition(c, status.ConditionTypeReady); rc != nil {
+			nc.Status = rc.Status
 			nc.Reason = rc.Reason
 			nc.Message = rc.Message
 		}
 
 	case operatorv1.Removed:
-		nc.Status = corev1.ConditionFalse
+		nc.Status = metav1.ConditionFalse
 		nc.Reason = string(operatorv1.Removed)
 		nc.Message = "Component ManagementState is set to " + string(operatorv1.Removed)
 
 	default:
 		return fmt.Errorf("unknown state %s ", s.GetManagementState(dsc))
 	}
 
-	conditionsv1.SetStatusCondition(&dsc.Status.Conditions, nc)
+	conditions.SetStatusCondition(dsc, nc)
```

Comment:
I was thinking in direction of access via Manager only

Reply:
Added a commit 0168d5a955a4d248a2aaa6b06ea7ad20e9313796 that goes in this direction.

Not 100% satisfied though, but it may require some more thinking and refactoring which could be done as a separate task.

Reply:
But do you share the attitude that the Manager wrapper with forcing ordering is usable for any access? If you think that having that middle level API is fine, I won't argue :)

Reply:
Yeah, here the thing is that the component specific "ready" condition, would contribute to the DSC's "ComponentsReady" condition, which then contribute to the top level ready conditions, so I need to find a good way to get the conditions manager supporting this case,